src/Admin/AdminConsole/Controllers/ProviderOrganizationsController.cs

```diff
@@ -60,7 +60,11 @@ public async Task<IActionResult> DeleteAsync(Guid providerId, Guid id)
             providerOrganization,
             organization);
 
-        await _removePaymentMethodCommand.RemovePaymentMethod(organization);
+        if (!string.IsNullOrEmpty(organization.GatewayCustomerId) &&
+            !string.IsNullOrEmpty(organization.GatewaySubscriptionId))
+        {
+            await _removePaymentMethodCommand.RemovePaymentMethod(organization);
+        }
```

Comment:
Should this be part of the command? Or are you trying to avoid calling a command from within a command?

Reply:
Yeah, originally I had `RemoveOrganizationFromProviderCommand` invoke `RemovePaymentMethodCommand`, but I was told not to do so when this was originally implemented.